samples/dnn/gpt2_inference.py

```diff
@@ -96,10 +101,22 @@ def gpt2_inference(net, tokens, max_length, num_return_sequences=5):
     num_return_sequences = args.batch_size
     prompt = args.prompt
 
+    net = cv.dnn.readNet(args.model)
+    input_token_size = net.getLayerShapes([], 0, 0)[0][0][1]
+
     enc = tiktoken.get_encoding('gpt2')
     tokens = enc.encode(prompt)
 
-    net = cv2.dnn.readNetFromONNX(args.model)
+    # Check if the prompt is of the same length as the input tokens
+    # if not, pad the tokens else truncate the tokens
+    if len(tokens) > input_token_size:
+        tokens = tokens[:input_token_size]
+    elif len(tokens) < input_token_size:
+        tokens2pad = input_token_size - len(tokens)
+        # append <space> token to the prompt
+        tokens += [220] * tokens2pad
```

Comment:
Not sure that whitespace is a correct pad token. There are also versions with EOS or PAD. Bit ok to merge with space now.

source: https://github.com/search?q=repo%3Ahuggingface%2Ftransformers+pad_token&type=code&p=2

Reply:
EOS I tried this. Your prompt followed by EOS make model output sentences that are not related to your prompt (as you indicatited end of context by EOS)